engine/pkg/cmd/master/master.go

```diff
@@ -57,6 +57,8 @@ func newOptions() *options {
 func (o *options) addFlags(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&o.masterConfig.Addr, "addr", o.masterConfig.Addr, "Set the listening address for server master")
 	cmd.Flags().StringVar(&o.masterConfig.AdvertiseAddr, "advertise-addr", o.masterConfig.AdvertiseAddr, "Set the advertise listening address for client communication")
+	cmd.Flags().StringVar(&o.masterConfig.AdvertiseLeaderAddr, "advertise-leader-addr", o.masterConfig.AdvertiseLeaderAddr,
```

Comment:
What happens if we always use the `advertise-addr`?

Reply:
It means executors must configure at least one valid address for servermaster. But it may be not always work well scenario.

For example:
1. The operator deploys a cluster with a single servermaster with `--advertise-addr=servermaster-0`.
2. All executor is configured to connect `servermaster-0`.
3. The operator scales up servermaster to 2 replicas. The new servermaster's address is `servermaster-1`.
4. servermaster-0 is down and servermaster-1 is elected to become the leader.
5. All executors will crash, because they don't know the servermaster-1's address.

Reply:
Why can the executor not detect the master topology? I think we can reference the interaction between TiDB and PD. TiDB will detect the latest PD information without the extra command line parameter.

Reply:
There are some drawbacks of the interaction between TiDB and PD. Like PD client, executors can also detect the master topology. But it can't persist the topology. Once it restarts, or it encounters the scenario as above, it may be unable to connect to master.

Reply:
> It means executors must configure at least one valid address for servermaster. But it may not always work well in all scenarios.
> 
> For example:
> 
> 1. The operator deploys a cluster with a single servermaster with `--advertise-addr=servermaster-0`.
> 2. All executor is configured to connect `servermaster-0`.
> 3. The operator scales up servermaster to 2 replicas. The new servermaster's address is `servermaster-1`.
> 4. servermaster-0 is down and servermaster-1 is elected to become the leader.
> 5. All executors will crash, because they don't know the servermaster-1's address.

@lonng This example can be also used for the interaction between TiDB and PD.